packages/svelte-vscode/docs/preprocessors/typescript.md

```diff
@@ -8,17 +8,40 @@ To tell us to treat your script tags as typescript, add a `type` or `lang` attri
     export let name: string;
 </script>
 
-<!-- Or add lang="typescript" -->
+<!-- Or add lang="typescript" or lang="ts" -->
 <script lang="typescript">
     export let name: string;
 </script>
 ```
 
-Now you'll need to add a `svelte.config.js` file at the root of your project to tell svelte how to convert your TypeScript into JavaScript that it understands.
+This will work with Svelte and TypeScript as long as you enable `svelte-preprocess` and `@rollup/plugin-typescript`:
 
-You likely already have this configuration somewhere if you are/are planning to use TypeScript with svelte, e.g. webpack config, rollup config, etc.
+- Install these packages `npm i svelte-preprocess typescript tslib @rollup/plugin-typescript`
+- Add these lines to `rollup.config.js`:
 
-_Tip: To avoid duplication of config, you can import the `svelte.config.js` file in your bundle configuration_
+```js
+// ...
+import sveltePreprocess from 'svelte-preprocess';
+import typescript from '@rollup/plugin-typescript';
+
+// ...
+  plugins: [
+    svelte({
+      // ...
+      preprocess: sveltePreprocess(), // <--
+    }),
+
+    // ...
+    commonjs(),
+    typescript(), // <-- added below commonjs
+    // ...
+```
+
+And this should work to enable full TypeScript checking in your Svelte files. For further discussion and a clonable template [see this issue](https://github.com/sveltejs/language-tools/issues/161).
+
+> Caveat: Your entry file (`main.js`) has still to be a javascript file
+
+You may optionally want to extract this out to a standalone `svelte.config.js` file - this used to be necessary but is no longer required. However you may wish to do this to avoid duplication of config.
```

Comment:
Hi, it might be worthwhile adding an example for this. At first read I got the impression that I could move the config to a `svelte.config.js` file and not even have to import it into the Rollup config. Or is that possible now? 
And nit-picky, but I would remove the "this used to be necessary bit" :)

Reply:
I updated the doc a little, see https://github.com/sveltejs/language-tools/blob/master/packages/svelte-vscode/docs/preprocessors/typescript.md . I removed both the "used to be" reference and the "you can deduplicate" info for brevity.

Reply:
Perfect! With that change, I think (hope) the user will know to refactor their rollup config if they use the svelte config file :)